components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -3330,15 +3330,18 @@ public static boolean jsFunction_updateApplication(Context cx,
            if (apps == null || apps.length == 0) {
                 return false;
             }
-            //check whether there is an app with same name
+
+            Map appsMap = new HashMap();
             for (Application app : apps) {
-                if (app.getName().equals(name)) {
-                    return false;
-                }
+                appsMap.put(app.getName(), app);
             }
 
             for (Application app : apps) {
-                if (app.getName().equals(oldName)) {
+                if(appsMap.containsKey(app.getName())){
```

Comment:
The first iteration reads the app names and put it into the map. (This happens before updating the apps)
And the second iteration checks whether the newName of the app (after update) is a name of already existing app by calling appsMap.containsKey(newName)
The code has been changed and the actual check in the code is if (app.getName().equals(oldName))
I have changed the name parameter to newName to reduce the confusion (Code improved in 8ecc7ed5684940f1f69ef8943428a017e8ab417d)